### PR TITLE
docs: Test PR guide

### LINKED ISSUE #542: Create guide on how to test PRs and backend with Swagger UI
## Description

As a contributor for testing,
I need a guide to help me understand how Swagger UI works and how to use multiple APIs from the project,
so that I can properly test pull requests from contributors that solve issues for the project.

Idea is to create a Markdown-based guide on the repository to help guide testers. [Conversation started here on Zulip](https://anitab-org.zulipchat.com/#narrow/stream/216325-quality-assurance/topic/PR.20that.20needs.20Testing/near/194472739).

_this has already an assignee_

### Update [Required]
- [ ] Create guide on docs folder

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by rpattath**
### Description
Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change.

Fixes #542 

### Type of Change:
- Documentation

### CODE REVIEW

File: docs/test-pr-guide.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+2. Check on Zulip chat if any of the PR's 
+[here](https://github.com/anitab-org/mentorship-backend/labels/Needs%20Testing) are high-priority if the priority is not set already.
+
+3. If the PR's are not priortized then pick any 1 PR from the list from the link in step 2 to test locally.
+
+4. Go to the Issue that PR is fixing and follow the steps to reproduce that issue while you are under the develop branch.
+
+5. Run the following commands to get to the PR branch
+
+git checkout -b <contributor>-<branch-name> develop
```
Comment (isabelcosta) [inline docs/test-pr-guide.md:16]: I would put this code inside "```" block code text, to appear like this:
```
git checkout -b <contributor>-<branch-name> develop
git pull https://github.com/<contributor>/mentorship-backend/ <branch-name>
```

File: docs/test-pr-guide.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+[README](https://github.com/anitab-org/mentorship-backend/blob/develop/README.md) to setup the system running locally.
+
+2. Check on Zulip chat if any of the PR's 
+[here](https://github.com/anitab-org/mentorship-backend/labels/Needs%20Testing) are high-priority if the priority is not set already.
+
+3. If the PR's are not priortized then pick any 1 PR from the list from the link in step 2 to test locally.
+
+4. Go to the Issue that PR is fixing and follow the steps to reproduce that issue while you are under the develop branch.
+
+5. Run the following commands to get to the PR branch
```
Comment (isabelcosta) [inline docs/test-pr-guide.md]:
```suggestion
5. Run the following commands to get to the PR branch, where `<contributor>` is the GitHub username of the contributor that submitted the PR:
```

File: docs/test-pr-guide.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+7. Reproduce the issue and test the fix.
+
+8. Get screenshots/gifs of before and after the fix and attach them to the PR comment.
+
+<h1>Steps to test multiple features (User Registration, List users, ...) of the Mentorship System.</h1>
+
+<h2>Registering users</h2>
+Register 3 users. Follow instructions in
+
```
Comment (isabelcosta) [inline docs/test-pr-guide.md:29]:
```suggestion
```

File: docs/test-pr-guide.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+[this demo on Youtube](https://www.youtube.com/watch?v=xRZrdR47R-w&feature=youtu.be&t=672) to register users. The users used here are
```
Comment (isabelcosta) [inline docs/test-pr-guide.md]: I would put this video demo, right after "Follow instructions in"

File: docs/test-pr-guide.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+<h1>Steps to test multiple features (User Registration, List users, ...) of the Mentorship System.</h1>
+
+<h2>Registering users</h2>
+Register 3 users. Follow instructions in
+
+[this demo on Youtube](https://www.youtube.com/watch?v=xRZrdR47R-w&feature=youtu.be&t=672) to register users. The users used here are 
+
+* testusera
+* testuserb
+* testuserc.
```
Comment (isabelcosta) [inline docs/test-pr-guide.md]:
```suggestion
* testuserc
```

File: docs/test-pr-guide.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+![Register user screenshot](https://user-images.githubusercontent.com/26095715/79673508-4a582180-81a8-11ea-87ce-fe0bfda82fea.png)
+
+<h2>Login as the users and capture the access tokens</h2>
+
+Steps to login and capture the token are shown in 
+
+[this video](https://www.youtube.com/watch?v=xRZrdR47R-w&feature=youtu.be&t=672)
```
Comment (isabelcosta) [inline docs/test-pr-guide.md:44]: "this video" right after "shown in"

File: docs/test-pr-guide.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
+<h1>Steps to test a PR</h1>
+
+1. Follow instructions in 
+
+[README](https://github.com/anitab-org/mentorship-backend/blob/develop/README.md) to setup the system running locally.
```
Comment (isabelcosta) [inline docs/test-pr-guide.md:4]: place it right after "Follow instructions in "

File: docs/test-pr-guide.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+2. Check on Zulip chat if any of the PR's 
+[here](https://github.com/anitab-org/mentorship-backend/labels/Needs%20Testing) are high-priority if the priority is not set already.
+
+3. If the PR's are not priortized then pick any 1 PR from the list from the link in step 2 to test locally.
```
Comment (isabelcosta) [inline docs/test-pr-guide.md]:
```suggestion
3. If the PRs are not prioritized then pick any 1 PR from the list from the link in step 2 to test locally.
```

**Review by isabelcosta — CHANGES_REQUESTED** [review]
I left some changes to make the markdown rendered doc easier to skin through, additionally I would make bold the text containing the users, when you mention them, I will comment in one of the cases.

Let me know what you think if you feel any of the changes I requested don't make sense ;)

@rpattath thank you so much for doing this document, this will be great for the QA process we are starting!

File: docs/test-pr-guide.md (hunk 4 above)
Comment (rpattath) [reply docs/test-pr-guide.md]: done

**Comment by rpattath** [comment]
@isabelcosta please have a look now

File: docs/test-pr-guide.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+<h3>Accept the mentorship request</h3>
+
+testuserb accepts the mentorship request from testusera.
+
+![Accepted request screenshot](https://user-images.githubusercontent.com/26095715/79673500-4926f480-81a8-11ea-9c97-bac8cc800f85.png)
+
+<h3>List current relations of a user</h3>
+
+List the current relationships of testusera. The relation should be state 2 (accepted).
```
Comment (isabelcosta) [inline docs/test-pr-guide.md]:
```suggestion
List the current relationships of _testusera_. The relation should be state 2 (accepted).
```
I forgot to mention this, having these names of user a little different to be easily skimmed through, what do you think? @rpattath

**Comment by isabelcosta** [comment]
@rpattath as a tip for the future, and maybe we have to improve the PR template to easily make that more clear, but I changed your markdown from :
```
Fixes # [Issue 542](https://github.com/anitab-org/mentorship-backend/issues/542)
```
To:
```
Fixes #542
```
this makes GitHub understand that once this PR is merged the issue mentioned will be closed. you can see this changed in this PR description.

File: docs/test-pr-guide.md (hunk 9 above)
Comment (isabelcosta) [reply docs/test-pr-guide.md]: Can you do this for everytime you mention in a sentence the username?

**Review by isabelcosta — APPROVED** [review]
This is great!
Thank you so much for this contribution 🎉 @rpattath

**Review by gaurivn — APPROVED** [review]

**Comment by sakshi1499** [comment]
@isabelcosta Should we not make it in one commit? Otherwise looks awesome!

**Comment by isabelcosta** [comment]
@sakshi1499 I can squash and merge, we have been talking in the community about this and maybe we should clarify a bit more. But in my opinion, I won't ask of contributors to squash, because sometimes creates confusion and problems with git history in contributors branch, so for me its good like this, and I will use the "Squash and merge" option, when merging.

**Comment by isabelcosta** [comment]
Can you leave your approval @sakshi1499 if all is good?

**Review by sakshi1499 — APPROVED** [review]
Great! I did read the discussion. :) @rpattath Awesome work!

**Comment by isabelcosta** [comment]
Thank you sooo much @gaurivn and @sakshi1499 for your reviews! 🎉

**Review by bartekpacia — COMMENTED** [review]
Nice job explaining it.

**Pull request merged**